chain/network/src/test_utils.rs

```diff
@@ -233,30 +233,28 @@ pub struct MockPeerManagerAdapter {
     pub notify: Notify,
 }
 
-impl MsgRecipient<WithSpanContext<PeerManagerMessageRequest>> for MockPeerManagerAdapter {
-    fn send(
+impl CanSendAsync<PeerManagerMessageRequest, Result<PeerManagerMessageResponse, ()>>
+    for MockPeerManagerAdapter
+{
+    fn send_async(
         &self,
-        msg: WithSpanContext<PeerManagerMessageRequest>,
-    ) -> BoxFuture<'static, Result<PeerManagerMessageResponse, MailboxError>> {
-        self.do_send(msg);
-        future::ok(PeerManagerMessageResponse::NetworkResponses(NetworkResponses::NoResponse))
-            .boxed()
+        message: PeerManagerMessageRequest,
+    ) -> BoxFuture<'static, Result<PeerManagerMessageResponse, ()>> {
+        self.requests.write().unwrap().push_back(message);
+        self.notify.notify_one();
+        async { Err(()) }.boxed()
```

Comment:
Does this mean that MockPeerManager doesn't support send_async? Shouldn't it rather be a pending() future?